packages/file-asset-apis/src/s3/s3-helpers.ts

```diff
@@ -15,7 +15,15 @@ import { S3ClientParams, S3ClientResponse, S3RetryRequest } from './client-types
 export async function s3RequestWithRetry(
     retryArgs: S3RetryRequest.GetObjectWithRetry,
     attempts?: number
-): Promise<S3ClientResponse.GetObjectOutput>
+): Promise<S3ClientResponse.GetObjectCommandOutput>
+export async function s3RequestWithRetry(
+    retryArgs: S3RetryRequest.PutObjectWithRetry,
+    attempts?: number
+): Promise<S3ClientResponse.PutObjectCommandOutput>
+export async function s3RequestWithRetry(
+    retryArgs: S3RetryRequest.DeleteObjectWithRetry,
+    attempts?: number
+): Promise<S3ClientResponse.DeleteObjectOutput>
```

Comment:
Could all of these function signatures be collapsed into one if we did:

```
S3RetryRequest.RetryArgs = RetryArgsA | RetryArgsB | RetryArgsC ...
S3RetryRequest.S3RetryResponse = OutputA | OutputB | OutputC ...
```

I don't know if that's better, but the only difference here is in the parameter and response types, right?